Library/ViewModels/ProjectPamphletViewModel.swift

```diff
@@ -224,3 +192,29 @@ private func cookieFrom(refTag: RefTag, project: Project) -> HTTPCookie? {
 
   return HTTPCookie(properties: properties)
 }
+
+private func fetchProjectAndLiveStreams(projectOrParam: Either<Project, Param>)
+  -> SignalProducer<(Project, [LiveStreamEvent]?), NoError> {
+
+    let param = projectOrParam.ifLeft({ Param.id($0.id) }, ifRight: id)
+
+    let projectAndLiveStreams = AppEnvironment.current.apiService.fetchProject(param: param)
+      .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+      .demoteErrors()
+      .flatMap { project -> SignalProducer<(Project, [LiveStreamEvent]?), NoError> in
+
+        AppEnvironment.current.liveStreamService
+          .fetchEvents(forProjectId: project.id, uid: AppEnvironment.current.currentUser?.id)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .flatMapError { _ in SignalProducer(error: SomeError()) }
+          .timeout(after: 5, raising: SomeError(), on: AppEnvironment.current.scheduler)
+          .materialize()
+          .map { (project, .some($0.value?.liveStreamEvents ?? [])) }
```

Comment:
Ah so this is really subtle and will cause some tracking tests to fail. We want to be explicit that this is an optional for the filtering logic to work when tracking the page view (i.e. don't track if LiveStreamEvents array is nil, wait until we have the full project data). Specifically on a timeout or when the load errors, we still want to track everything so we have to coalesce here. Make sense?